service-base/src/test/java/org/eclipse/hono/service/metric/MicrometerBasedMetricsTest.java

```diff
@@ -205,6 +206,24 @@ public void testPayloadSizeForCommandMessages(final MeterRegistry registry) {
                 registry.find(MicrometerBasedMetrics.METER_COMMANDS_PAYLOAD).summary().totalAmount());
     }
 
+    /**
+     * Verifies that the connection time duration is recorded for the given tenant when devices get connected and
+     * disconnected.
+     *
+     * @param registry The registry that the tests should be run against.
+     */
+    @ParameterizedTest
+    @MethodSource("registries")
+    public void testConnectionTimeDuration(final MeterRegistry registry) {
+        final MeterRegistry meterRegistry = Mockito.spy(registry);
```

Comment:
The whole point of a parameterized test is that you use the passed in object(s), isn't it? FMPOV you can verify if the meters have been updated correctly in the same way as in the existing test cases ...

Reply:
Ofcourse the passed in object is already being used ;-) 
As per my understanding, only a spy or a mock object can be used within the `verify` method.
Hence here a spy of the passed in object namely `meterRegistry` is created and it is used in `verify(meterRegistry, times(1)).timer(...` to verify if the `timer(..)` is being invoked.

Reply:
Oh, my bad. I didn't notice. In any case, you nee to [be careful with spying on real objects](https://javadoc.io/doc/org.mockito/mockito-core/2.28.2/org/mockito/Mockito.html#13). I this case it should be ok though, as the *timer* method is not final.